### ISSUE TITLE
AI (chatgpt / Llama) integration

### ISSUE THREAD

**Issue by snickers2k** [linkwarden/linkwarden#396]
**Is your feature request related to a problem? Please describe.**
- categorizing / tagging takes time (especially with a lot of categories)
- remembering what the bookmark was, sometimes needs to open the page again to take an overview

**Describe the solution you'd like**
- would be awesome to have chatgpt or something else reading the site and giving a quick summery and ideally categorize them correctly + giving some tags
-- maybe even utilizing a chatgpt-plugin to search bing/google for other similar topics and refer them

**Comment by mmartinortiz** [comment]
I have also been thinking about this. This could be possible with:
- The archived JSON format, where the text is saved in a clean way
- [LlamaIndex](https://github.com/run-llama/llama_index), for creating an index with the archived links
- A query for an LLM model based on the user question (like "provide tags for this text", or "summarize the information about X/Y")

The LLM model could be a remote one, like OpenAI, or a local one, like those provided by [llama-file](https://github.com/mozilla-ocho/llamafile).

This could be used both for providing categories, tags, etc. Or searching for information based on the archived sites.

**Comment by e-p-s** [comment]
Local AI is a must have. There are many projects that offer an api. ollama, localai.cpp ans so on

**Comment by vanBassum** [comment]
What about using AI to search for a link? So using the chat to 'I should have saved a URL about an article containing ....' something similar to 'pieces for developers'.

**Comment by daniel31x13** [comment]
Soon!

**Comment by SomeRandomCpu** [comment]
Yes! Hoarder has this feature, and it's the only thing keeping me from switching from raindrop.io

**Comment by arcb01** [comment]
Really looking forward to this!

**Comment by daniel31x13** [comment]
Fully local Ollama-based AI tagging Released in [v2.9.0](https://github.com/linkwarden/linkwarden/releases), enjoy!

*P.S. for those who gave a thumbs down, this feature is totally optional and if you don't specify the necessary environment variables, you won't even see the word "AI" in the app...*

**Comment by truppelito** [comment]
I'm sorry for slightly hijacking this thread, but I'd just like to ask: once local AI is configured, is it possible to tell Linkwarden to tag already existing websites (as opposed to only doing it for the newly inserted ones)?

**Comment by mmartinortiz** [comment]
> I'm sorry for slightly hijacking this thread, but I'd just like to ask: once local AI is configured, is it possible to tell Linkwarden to tag already existing websites (as opposed to only doing it for the newly inserted ones)?

This issue #989 asks the same question

**Comment by bunkerman** [comment]
Do you also implement to use an OpenAI compatible API endpoint(e.g. localAI)?

**Comment by daniel31x13** [comment]
> is it possible to tell Linkwarden to tag already existing websites

Next release 🔜

> Do you also implement an option to use an OpenAI compatible API endpoint (e.g. localAI)?

Also next release 🔜

**Comment by firestrife23** [comment]
> > Do you also implement an option to use an OpenAI compatible API endpoint (e.g. localAI)?
> 
> Also next release 🔜

next release??..... how.... much... longer... 😢 actually I'm exciting for Linkwarden + LiteLLM!
